src/platform.ts

```diff
@@ -1,33 +1,50 @@
 import { APIEvent } from 'homebridge';
 import type { API, DynamicPlatformPlugin, Logger, PlatformAccessory, PlatformConfig } from 'homebridge';
 
-import { PLATFORM_NAME, PLUGIN_NAME } from './settings';
-import { SubDevice, DeviceType } from './models';
-import { SwitchAccessory } from './switchAccessory';
-import { EnergenieApi } from './energenieApi';
-import { MiHomePlatformAccessory } from './platformAccessory';
+import { PLATFORM_NAME, PLUGIN_NAME, MIHOME_API_BASE_URL } from './settings';
+import { SwitchAccessory } from './accessories/switchAccessory';
+
+import { DeviceType } from './energenieApi/models';
+import { EnergenieApi } from './energenieApi/energenieApi';
 
 /**
- * HomebridgePlatform
+ * MiHome Gateway Platform
  * This class is the main constructor for your plugin, this is where you should
  * parse the user config and discover/register accessories with Homebridge.
  */
-export class MiHomeGateway implements DynamicPlatformPlugin {
+export class MiHomeGatewayPlatform implements DynamicPlatformPlugin {
   public readonly Service = this.api.hap.Service;
   public readonly Characteristic = this.api.hap.Characteristic;
-  public readonly EnergenieApi: EnergenieApi;
 
   // this is used to track restored cached accessories
   public readonly accessories: PlatformAccessory[] = [];
 
+  public readonly EnergenieApi: EnergenieApi;
+
   constructor(
     public readonly log: Logger,
     public readonly config: PlatformConfig,
     public readonly api: API,
   ) {
     this.log.debug('Finished initializing platform:', this.config.name);
 
-    this.EnergenieApi = new EnergenieApi(log, config.username, config.password, config.baseUrl || 'https://mihome4u.co.uk/api/v1/');
+    if (!this.config.username) {
+      this.log.error('Username not set.');
+    }
+
+    if (!this.config.password && !this.config.token) {
+      if (this.config.token) {
```

Comment:
This will never be true as we've confirmed about that `this.config.token` == false